handbook/repository/nx-tags.md

```diff
@@ -0,0 +1,96 @@
+
+It can be just one tag which is applied to both the application project and all of the associated library projects. Or there can be multiple application tags with different rules to segments the library projects.
+
+| Tag               | Description             | Can depend on                                        | Can be depended on by          |
+| ----------------- | ----------------------- | ---------------------------------------------------- | ------------------------------ |
+| `api`             | API domain project.     | `js`, `node`, `nest`, `api`                          | Other API projects.            |
+| `portals-mypages` | Service portal project. | `js`, `dom`, `react`, `react-spa`, `portals-mypages` | Other service portal projects. |
+| `portals-admin`   | Admin portal project.   | `js`, `dom`, `react`, `react-spa`, `portals-admin`   | Other admin portal projects.   |
+
+## Tag prefixes
+
+The above tag **must** be prefixed by one or both of the following prefixes to opt projects into "Can depend on" and "Can be depended on by" rules.
+
+- `scope:` - by applying a scope tag to a project, you are saying that ESLint should enforce project boundaries for that project using the "Can depend on" rules documented above.
+- `lib:` - lib tags describe what kind of project it is to enforce the "Can be depended on by" rules documented above.
+
+This may be confusing, but basically, every project should have a lib tag, and eventually every project should also have a scope tag. You should use both prefixes in projects you maintain, and at least add tags with the lib prefix in projects you depend on.
+
+This is to help us migrate to project boundaries over time. When we have added lib and scope tags to all projects, we can get rid of the lib prefix.
+
+### Examples
+
+Let's say you want to enforce project boundaries in project X which is an existing NextJS project you maintain. It depends on NextJS library Y which you maintain and React library Z which you don't maintain.
+
+First step: Open `apps/x/project.json` and add the following tag:
+
+```json
+{
+  "tags": ["scope:next"]
+}
+```
+
+This marks the project as a NextJS project, and tells ESLint to verify its imports. At this point ESLint would complain that project X can't import libraries Y and Z.
+
+{% hint style="info" %}
+You should not list "lib:next" in projects under the "app" folder since they should never be depended on by other projects.
+{% endhint %}
+
+Next, open `libs/y/project.json` and add the following tags:
+
+```json
+{
+  "tags": ["scope:next", "lib:next"]
+}
+```
+
+The `lib:next` tag allows the library to be imported by other projects with the "scope:next" tag (fixing one of the ESLint errors in project X). Again, adding `scope:next` configures ESLint to also verify the imports of library Y. It might find some boundary errors which you should fix.
```

Comment:
```suggestion
The `lib:react-next` tag allows the library to be imported by other projects with the "scope:react-next" tag (fixing one of the ESLint errors in project X). Again, adding `scope:react-next` configures ESLint to also verify the imports of library Y. It might find some boundary errors which you should fix.
```